src/cloudant/database.py

```diff
@@ -329,16 +329,21 @@ def get_view_result(self, ddoc_id, view_name, raw_result=False, **kwargs):
         else:
             return view.result
 
-    def create(self):
+    def create(self, **kwargs):
         """
         Creates a database defined by the current database object, if it
         does not already exist and raises a CloudantException if the operation
         fails.  If the database already exists then this method call is a no-op.
 
+        :param bool none_on_exists: Boolean flag dictating whether or
+            not to return none object when the database already exists.
         :returns: The database object
         """
         if self.exists():
-            return self
+            if kwargs.get('none_on_exists', True):
+                return None
+            else:
+                return self
```

Comment:
I think that passing back `None` if a database exists is an odd behavior to introduce to fix this bug.  Instead I propose that you add `throw_on_exists` to the signature of the database `create` method and default it to `False` so as not to change the current default behavior of this method call.  Like `def create(self, throw_on_exists=False):`.  Then I would change the `if self.exists():` check to `if not throw_on_exists and self.exists():`.  Again preserving the current behavior of the method.  With the logic changed to this the method can now throw a CloudantDatabaseException 409 which we can then specifically catch in the client `create_database` method and re-wrap as a CloudantClientException.  Once again preserving the current behavior of that `create_database` method.  Obviously we will also need to change the client `create_database` logic to call `new_db.create(kwargs.get('throw_on_exists', False))` and remove the `new_db.exists()` if logic dance in order to complete the fix.

This allows us to:
- Preserve current client and database object functionality which means a micro release.
- Remove the extra call to `exists()` bug.